### PR TITLE
TGBOT-103 create UserService

### PR DESCRIPTION
**Pull request by yurasava**
https://schneidervs.atlassian.net/jira/software/projects/TGBOT/boards/2?selectedIssue=TGBOT-103

UserService created without tests

### CODE REVIEW

File: stoservice/src/main/java/com/tutrit/stoservice/service/UserService.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    public UserService(final UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public User saveUser(User user) {
+        return userRepository.save(user);
+    }
+
+    public void setUserRepository(UserRepository userRepository) {
```
Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/service/UserService.java:18]: You don't have a default constructor, so it's not a necessary method, when you create a class you already set a value in the class field.

File: stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java [hunk 2]
```diff
@@ -16,29 +14,26 @@ class UserRepositoryTest {
...
+        userRepository.save(new User("4", "user4", "phone4"));
     }
 
     @Test
     void createUser() {
+        int expectedCount = userRepository.count() + 1;
         userRepository.save(new User("5", "user5", "phone5"));
-        assertEquals(5, UserRepository.userMap.size());
-        assertTrue(UserRepository.userMap.containsKey("5"));
+        assertEquals(expectedCount, userRepository.count());
```
Comment (schneidervs) [inline stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java:27]: This is a strange test in my opinion. If you are testing a method, why are you testing it with another method that has not yet been tested? Whereas your method returns a value.
Maybe you shouldn't make the repository private?

**Review by ASanovS — APPROVED** [review]

File: stoservice/src/main/java/com/tutrit/stoservice/context/ApplicationContextLoader.java [hunk 3]
```diff
@@ -13,17 +17,27 @@ private ApplicationContextLoader() {
...
         put(NotACommandController.class, new NotACommandController());
+        put(UserController.class, new UserController(get(UserService.class)));
+
+        Map<Command, CommandController> commandCommandControllerMap = Map.of(
+                get(DataController.class).getCommand(), get(DataController.class),
+                get(ExitController.class).getCommand(), get(ExitController.class),
+                get(HelpController.class).getCommand(), get(HelpController.class),
+                get(NotACommandController.class).getCommand(), get(NotACommandController.class),
+                get(UserController.class).getCommand(), get(UserController.class)
+        );
```
Comment (AlexanderYarokhovich) [inline stoservice/src/main/java/com/tutrit/stoservice/context/ApplicationContextLoader.java:40]: without it?

File: stoservice/src/main/java/com/tutrit/stoservice/controller/Command.java [hunk 4]
```diff
@@ -1,12 +1,11 @@
 package com.tutrit.stoservice.controller;
 
-import java.util.*;
-
 public enum Command {
 
     SHOW_DATA("show data"),
     EXIT("exit"),
     HELP("help"),
+    REGISTER_NEW_USER("new_user"),
```
Comment (AlexanderYarokhovich) [inline stoservice/src/main/java/com/tutrit/stoservice/controller/Command.java:8]: add a space like in "show data"

File: stoservice/src/main/java/com/tutrit/stoservice/controller/RequestDispatcher.java [hunk 5]
```diff
@@ -4,20 +4,22 @@
...
 
-    public static Logger logger = LoggerFactory.getLogger("Dispatcher");
+    public static final Logger logger = LoggerFactory.getLogger("Dispatcher");
 
     public void doDispatch(String userInput) {
         Request request = new Request(userInput);
         Response response = new Response();
 
-        switch (Command.fromString(request.getCommand())) {
+        switch (Command.fromString(request.getCommand().split(" -data")[0])) {
```
Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/controller/RequestDispatcher.java:17]: How about making it a method or a class?

**Review by schneidervs — APPROVED** [review]

File: stoservice/src/main/java/com/tutrit/stoservice/controller/RequestDispatcher.java (hunk 5 above)
Comment (max-java) [inline stoservice/src/main/java/com/tutrit/stoservice/controller/RequestDispatcher.java:17]: It is more convenient to have util method for that

File: stoservice/src/main/java/com/tutrit/stoservice/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    private static final Command command = Command.REGISTER_NEW_USER;
+    private final UserService userService;
+
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public void doCommand(Request request, Response response) {
+        userService.saveUser(ParseUser.parseCommand(request));
```
Comment (max-java) [inline stoservice/src/main/java/com/tutrit/stoservice/controller/UserController.java:17]: rename parse command that way that it make clear that it creates an object

File: stoservice/src/test/java/com/tutrit/stoservice/provider/UserProviderTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        initialUsers.add(new User("Id0", "User0", "Phone0"));
+        initialUsers.add(new User("Id1", "User1", "Phone1"));
+        initialUsers.add(new User("Id2", "User2", "Phone2"));
+        initialUsers.add(new User("Id3", "User3", "Phone3"));
+        initialUsers.add(new User("Id4", "User4", "Phone4"));
+        initialUsers.add(new User("Id5", "User5", "Phone5"));
+        initialUsers.add(new User("Id6", "User6", "Phone6"));
+        initialUsers.add(new User("Id7", "User7", "Phone7"));
+        initialUsers.add(new User("Id8", "User8", "Phone8"));
+        initialUsers.add(new User("Id9", "User9", "Phone9"));
```
Comment (max-java) [inline stoservice/src/test/java/com/tutrit/stoservice/provider/UserProviderTest.java:27]: use provider

**Review by schneidervs — APPROVED** [review]

**Review by Sav4ykVladzislau — APPROVED** [review]

**Review by AlexanderYarokhovich — APPROVED** [review]

**Pull request merged**